### PR TITLE
Replacing PyMC3 plots w/ Arviz plots & sigma Param change [Part 7]

### PR DESCRIPTION
**Pull request by CloudChaoszero**
# Description
The following is a large PR breakdown of PR #16.

Replace PyMC3 dependent plots with arviz plots in case studies & examples.
* The issue is a 3-part step for the following:
  * Initiative via Issue https://github.com/pymc-devs/pymc3/issues/4371 (Leave open)

Replace parameter `sd` with `sigma` (e.g. some examples have `pm.Normal(...sd=...)`
* Closes issue #2

### CODE REVIEW

File: examples/samplers/MLDA_variance_reduction_linear_regression.ipynb [hunk 1]
```diff
@@ -533,7 +533,7 @@
     }
```
Comment (twiecki) [inline examples/samplers/MLDA_variance_reduction_linear_regression.ipynb:533]: <p>Should probably add the model context to this and below to future proof this and get rid of the annoying warning.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Comment by twiecki** [comment]
Can you try updating PyMC3 and Theano? Otherwise this seems like a Aesara bug. CC @brandonwillard

**Comment by CloudChaoszero** [comment]
@brandonwillard  Thanks for the response! Hmm, unfortunately I get some more errors. I updated the particular notebook to show the error, seen below.

(If the details are too much for this PR, I can submit a detailed issue in Aesara? Let me know!)
________

Adding these params in pm.sample()
```python
    method_names.append("Metropolis")
    traces.append(
        pm.sample(
            draws=ndraws,
            step=step_metropolis,
            chains=nchains,
            tune=nburn,
            discard_tuned_samples=discard_tuning,
            random_seed=sampling_seed,
            start=MAP,
            mp_ctx="forkserver",
            pickle_backend='dill',
        )
```
I get 
Screen Shot 2021-02-04 at 3 03 07 AM
due to dill pickle'ing. Well, I suspect that is the reason.

If I do not add the params
```
            mp_ctx="forkserver",
            pickle_backend='dill',
```

then I get a chained 0 error 😢

**Comment by brandonwillard** [comment]
Please, provide the complete trace as formatted markdown.  If the trace is too large, it can be wrapped in HTML `<details>` tags (e.g. [here](https://gist.github.com/joyrexus/16041f2426450e73f5df9391f7f7ae5f)).

**Comment by CloudChaoszero** [comment]
@brandonwillard Thanks for that!

Here is the Gist for when we use 
```
            mp_ctx="forkserver",
            pickle_backend='dill',
```

https://gist.github.com/CloudChaoszero/2ffbc8efa6157786102c8a4659bf5c11 

________

Here is the Gist for when we do not use the params, originally in the notebook 
https://gist.github.com/CloudChaoszero/6078a62bad062cc9ab541e150c914d49

**Comment by brandonwillard** [comment]
I don't see any connections to Theano-PyMC, but a quick search did yield [this open `dill` issue](https://github.com/uqfoundation/dill/issues/332).

**Comment by CloudChaoszero** [comment]
> I don't see any connections to Theano-PyMC, but a quick search did yield [this open `dill` issue](https://github.com/uqfoundation/dill/issues/332).

@brandonwillard Thanks for checking it out. After some further research, I founda. workaround is to use `cores` param to help out with this. I found this workaround from the [PyMC3 Discourse, found here.](https://discourse.pymc.io/t/problem-with-the-getting-started-with-pymc3-tutorial/2347/3)

Here is what I added at the end of each `pm.sample()`:
cores != None...i.e.
```python
            cores=1,
            mp_ctx="forkserver",

```

File: examples/samplers/MLDA_variance_reduction_linear_regression.ipynb (hunk 1 above)
Comment (CloudChaoszero) [reply examples/samplers/MLDA_variance_reduction_linear_regression.ipynb]: <p>Sounds great. I quite didn't understand what the FutureWarning recommended, but I am happy to update the description as a PR. </p><p> </p><p>Here was my take on what the FutureWarning recommended haha:</p><p> </p><pre class="ql-syntax" spellcheck="false">with model:
&nbsp; &nbsp; trace2_az = az.from_pymc3(trace2)
az.summary(trace2_az)
</pre><p> </p>

**Comment by CloudChaoszero** [comment]
As a note, I updated my package dependencies and did more research this time on navigating the errors encountered with pymc3. 

In particular, this PR uses `cores=1` in `pm.sample()` and some other small changes. 
* Though the perf runs long, it was successfully. 😄 

<details><summary>Deps</summary>

```
pymc3                     3.11.0                    dev_0    <develop>
theano-pymc               1.1.2                    pypi_0    pypi
```

</details>

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Thanks @CloudChaoszero!

**Pull request merged**